plugins/chat/assets/javascripts/discourse/models/chat-channel.js

```diff
@@ -119,6 +119,10 @@ export default class ChatChannel extends RestModel {
     return this.currentUserMembership.following;
   }
 
+  get canJoin() {
+    return this.currentUserMembership.meta?.can_join_chat_channel;
```

Comment:
I don't think this will work, since `current_user_membership` is not always present. We do set a default here:

https://github.com/discourse/discourse/blob/2c948d722fed3703ebd3a9faded775b5ae0bcd13/plugins/chat/assets/javascripts/discourse/models/chat-channel.js#L165-L172

But that does not have `meta.can_join_chat_channel`

Reply:
It's probably better to put `can_join_chat_channel` on `ChatChannelSerializer` `meta` instead, since then the membership doesn't have to exist.

Reply:
After having a bit more time with it. Yes, it makes much more sense to add it to the chant `ChatChannelSerializer`